### DISCUSSION TITLE
Rich Text Editor Example

### DISCUSSION THREAD

**Discussion by hafezrouzati** [emilk/egui#2806] [General]
I thought I remembered seeing a rich text editor with a menu bar for adding properties like Bold or Underline, etc. or changing the fonts of the text inside the TextEdit, but I can't seem to find the example now.  Can someone provide a pointer to the example?

Also are there any demos of file drag 'n drop?

Thanks emilk for this amazing library.  Currently building my first Rust app using egui and its been very interesting so far.

**Comment by lost22git** [comment]
[a code editor demo](https://github.com/emilk/egui/blob/master/crates/egui_demo_lib/src/demo/code_editor.rs)

[drag and drop demo](https://github.com/emilk/egui/blob/master/crates/egui_demo_lib/src/demo/drag_and_drop.rs)

**Comment by lost22git** [comment]
you can find others in https://www.egui.rs/#demo

**Comment by hafezrouzati** [comment]
Hi, thanks for your reply.

Yes I am aware of those examples, I was specifically looking for something that deals with the DroppedFile struct https://docs.rs/egui/0.21.0/egui/struct.DroppedFile.html

Also, was looking for something about rich text editing and how to manage the updates to the rich text when trying to add editing tools such as making bold, underline, changing fonts etc (I'm already using the code editor for rendering code).

Nonetheless, I appreciate your response.

**Reply by lost22git** [reply]
> Also, was looking for something about rich text editing and how to manage the updates to the rich text when trying to add editing tools such as making bold, underline, changing fonts etc (I'm already using the code editor for rendering code).

I guess it can be done like this：get the [CursorRange](https://docs.rs/egui/0.21.0/egui/widgets/text_edit/struct.CursorRange.html) of selected text and update [LayoutJob](https://docs.rs/egui/0.21.0/egui/text/struct.LayoutJob.html) cache.

**Reply by hafezrouzati** [reply]
Amazing, thanks for the pointer!
